apis/network/v1beta1/ipset_webhook.go

```diff
@@ -74,3 +133,124 @@ func (r *IPSet) ValidateDelete() error {
 	// TODO(user): fill in your validation logic upon object deletion.
 	return nil
 }
+
+// valiateIPSetNetwork
+// - networks are uniq in the list
+// - networks and subnets exist in netcfg
+// - FixedIP is a valid IP address
+// - FixedIP has correct IP version of subnet
+// - FixedIP is in the subnet cidr
+// - Route is only specified on a single network
+func valiateIPSetNetwork(
+	networks []IPSetNetwork,
+	path *field.Path,
+	netCfgSpec *NetConfigSpec,
+) field.ErrorList {
+	allErrs := field.ErrorList{}
+	netNames := map[string]field.Path{}
+	defaultRouteCount := 0
+
+	for _netIdx, _net := range networks {
+		path := path.Child("networks").Index(_netIdx)
+
+		// validate uniqe networks in request
+		allErrs = append(allErrs, valiateUniqElement(netNames, string(_net.Name), path, "name", errDupeNetworkName)...)
+
+		// validate if requested network is in NetConfig
+		f := func(n Network) bool {
+			return n.Name == _net.Name
+		}
+		netIdx := slices.IndexFunc(netCfgSpec.Networks, f)
+
+		if netIdx >= 0 {
+			// validate if requested subnet is in NetConfig
+			f := func(n Subnet) bool {
+				return n.Name == _net.SubnetName
+			}
+			subNetIdx := slices.IndexFunc(netCfgSpec.Networks[netIdx].Subnets, f)
+
+			if subNetIdx >= 0 {
+				// net and subnet are valid
+				// validate the requested FixedIP
+				if _net.FixedIP != nil {
+					cidr := netCfgSpec.Networks[netIdx].Subnets[subNetIdx].Cidr
+					_, ipPrefix, ipPrefixErr := net.ParseCIDR(cidr)
+					if ipPrefixErr != nil {
+						// this should never happen as the subnet CIDR was already validated
+						// via the netcfg webhook
+						allErrs = append(allErrs, field.Invalid(path.Child("cidr"), cidr, errInvalidCidr))
+						return allErrs
+					}
+
+					path := path.Child("fixedIP")
+					if err := valiateAddress(*_net.FixedIP, ipPrefix, path); err != nil {
+						allErrs = append(allErrs, err...)
+					}
+				}
+
+				// check that there are not multiple have the defaultRoute flag
```

Comment:
We need to allow one for IPv4 and one for IPv6. Not critical for now, can be a follow up.

Reply:
I have added this in the latest version I pushed